src/places_nearby.rs

```diff
@@ -142,6 +145,13 @@ impl Display for BadPlacesNearby {
 
 impl std::error::Error for BadPlacesNearby {}
 
+lazy_static! {
+    static ref COORD_COLON_REGEX: Regex =
+        Regex::new(r"^([-+]?[0-9]*\.?[0-9]*):([-+]?[0-9]*\.?[0-9]*)$",).unwrap();
+    static ref COORD_SEMICOLON_REGEX: Regex =
+        Regex::new(r"^([-+]?[0-9]*\.?[0-9]*);([-+]?[0-9]*\.?[0-9]*)$",).unwrap();
```

Comment:
How about doing this instead?
```suggestion
    static ref COORD_REGEX: Regex =
        Regex::new(r"^([-+]?(?:\d*\.)?\d+)[:;]([+-]?(?:\d*\.)?\d+)$",).unwrap();
```
A few things happen in here. First, I replaced the match with `:` with a match with `[:;]` in the middle, to support the new use case as I understand. I don't think you need 2 regular expression, one should be enough.

And also, see https://stackoverflow.com/a/12643073/7447059 for explanations about matching a float... Note that I didn't completely copy the proposition of StackOverflow since it adds a capturing group which is annoying for us. So I've added non-capturing groups with `?:` (see https://docs.rs/regex/latest/regex/index.html#grouping-and-flags).

Reply:
The thing is I have two distinct cases : either my coord is given as the string "coord:lat:lon" or "lat;lon".
With a single regex, I will also accept "coord:lat;lon" or "lat:lon", right ?

Reply:
OK, I missed [that part](https://github.com/hove-io/loki/pull/396/files#diff-bb021e8bd77967e936b82af8478d98c8a729235c3dba7a411e0953c86195b35fR170). And indeed, we might be able to write a single correct regular expression for your case, but at what price :rofl: Keeping 2 distinct regular expressions will probably be easier.

That said, why don't you put `coord:` directly in the regular expression to make it more explicit why you do have 2 regular expressions? Also, you will need one less `if` below if I understand correctly?